pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -65,11 +67,17 @@ func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
 		if namespace.Labels == nil {
 			namespace.Labels = map[string]string{}
 		}
-		if label, ok := namespace.Labels["maistra.io/ignore-namespace"]; !ok || label != "ignore" {
-			r.Log.Info("Adding maistra.io/ignore-namespace=ignore label to Request.Namespace")
-			namespace.Labels["maistra.io/ignore-namespace"] = "ignore"
-			err = r.Client.Update(context.TODO(), namespace)
+
+		// Label the control plane namespace so that Kiali knows about it
+		r.Log.Info(fmt.Sprintf("Adding %s label to namespace %s", common.MemberOfKey, r.Instance.Namespace))
+		namespace.Labels[common.MemberOfKey] = r.Instance.Namespace
+
+		if label, ok := namespace.Labels[common.IgnoreNamespace]; !ok || label != "ignore" {
+			r.Log.Info(fmt.Sprintf("Adding %s=ignore label to namespace %s", common.IgnoreNamespace, r.Instance.Namespace))
```

Comment:
this test prevents updating the namespace if the label already exists.  this code will be executed for every update to the CR, so we save an api call.

Reply:
Thanks for the clarification